pkg/matcher/annotation_matcher.go

```diff
@@ -28,8 +28,16 @@ const (
 func branchMatch(prunBranch, baseBranch string) bool {
 	// If we have targetBranch in annotation and refs/heads/targetBranch from
 	// webhook, then allow it.
+	// Also match for other variables which may exist between refs/heads and targetBranch
+	// ex: refs/heads/abc/xyz/targetBranch
+	// In those cases branch matching should fail
 	if filepath.Base(baseBranch) == filepath.Base(prunBranch) {
```

Comment:
do you think we need this if ?

Reply:
i think simply this will be enough
```
baseBranchValues := strings.Split(baseBranch, "refs/heads")
if baseBranchValues[1] == prunBranch {
return true
}
```

Reply:
also assume the case if `refs/heads` is not there in the name, then this will fail

Reply:
we need to do simple like - if there is ref/head in start, we need to remove that prefix for both names, and then the remaining string would match.

Reply:
> do you think we need this if ?

Yes its required for push scenario where no refs/heads specified in targetbranch but the payload coming for push always have appended refs/heads/{branch_name} and filepath.Base returns the last element of the path ignoring the `refs/heads` so to compare branchname it will be helpful

ex:

```
	fmt.Println("On Unix:")
	fmt.Println(filepath.Base("/foo/bar/baz.js"))
	fmt.Println(filepath.Base("/foo/bar/baz"))
	fmt.Println(filepath.Base("/foo/bar/baz/"))
	fmt.Println(filepath.Base("dev.txt"))
	fmt.Println(filepath.Base("../todo.txt"))
	fmt.Println(filepath.Base(".."))
	fmt.Println(filepath.Base("."))
	fmt.Println(filepath.Base("/"))
	fmt.Println(filepath.Base(""))

	Output:
	//
	On Unix:
	baz.js
	baz
	baz
	dev.txt
	todo.txt
	..
	.
	/
	.
```

But the aboue if condition created issue when there is something inbetween `refs/heads` and targetbranchname as explained in the comment section

Reply:
> we need to do simple like - if there is ref/head in start, we need to remove that prefix for both names, and then the remaining string would match.

actually i had this same thought but `filepath.Base` does the same operation so kept that condition without changing

Reply:
i mean the case where prunBranch also contains `ref/heads` is not handled, also the first if condition is of no use after this implementation

Reply:
Actually it will not fail instead go for checking another condition and succeed

But i rechecked the code and found that removing 

```
	if filepath.Base(baseBranch) == filepath.Base(prunBranch) {
		return true
	}
```
will work because we have regex comparison in later section to match the branch name